consensus/hotstuff/eventloop/event_loop.go

```diff
@@ -33,7 +33,7 @@ var _ component.Component = (*EventLoop)(nil)
 // NewEventLoop creates an instance of EventLoop.
 func NewEventLoop(log zerolog.Logger, metrics module.HotstuffMetrics, eventHandler hotstuff.EventHandler, startTime time.Time) (*EventLoop, error) {
 	proposals := make(chan *model.Proposal)
-	quorumCertificates := make(chan *flow.QuorumCertificate)
+	quorumCertificates := make(chan *flow.QuorumCertificate, 1)
```

Comment:
With single node cases we are running into deadlock
event loop submits a proposal -> proposal contains a vote -> one vote is enough to create a QC -> qc is created -> submitted to event loop -> event loop tries to send into unbuffered channel -> deadlock

Reply:
Is there a test case that captures the deadlock?

Another option is to [submit to even loop](https://github.com/onflow/flow-go/blob/yurii/5865-fixing-deactivated-tests/consensus/hotstuff/integration/instance_test.go#L339) with a go routine, so that the previous call can finish first. This would be actually close to the real case where the vote processor is processing on a different thread, and the QC is always built and submit from a different thread.

If we had a test case that captures this deadlock, you can try the alternative and verify if the test case could still pass.